src/nvim/ex_getln.c

```diff
@@ -767,7 +767,7 @@ static uint8_t *command_line_enter(int firstc, int count, int indent, bool clear
   }
 
   setmouse();
-  ui_cursor_shape();               // may show different cursor shape
+  setcursor();
```

Comment:
Instead, move the cursor back to curwin when leaving the cmdline (and remove duplicate call to `ui_cursor_shape()`).

Reply:
Is removing duplicate call to `ui_cursor_shape()` really necessary? It doesn't do anything if editor state hasn't changed.

Reply:
No but why would we keep it? It's called in setmouse() in the line above.

Reply:
Oh, I thought you were talking about `command_line_check()`.